addons/analytic/models/analytic_line.py

```diff
@@ -51,6 +51,29 @@ def _get_plan_fnames(self):
         project_plan, other_plans = self.env['account.analytic.plan']._get_all_plans()
         return [fname for plan in project_plan + other_plans if (fname := plan._column_name()) in self]
 
+    def _get_analytic_accounts(self):
```

Comment:
`_get_analytic_account_ids()` is defined in the analytic distribution mixin ('analytic.mixin'), here we implement it for the analytic plan mixin ('analytic.plan.fields.mixin'), so I'm not sure we can reuse what you suggest

Reply:
But why are we not using a distribution? with `analytic.mixin`?
Not sure why we should restrict to having only one account at 100% per plan

Reply:
It was decided to use the mixin plan to make the configuration simple. And it's going to be complicated to calculate the profitability of the project if we have to take into account distribution analytics. But maybe in the future we will change it to the distribution mixin if we find a way to make it work with the project profitiability.

Reply:
The profitability is computed based on `account_id` or `analytic_account_id`, right?
Which in the distribution would translated to a mandatory plan, that has to be 100%; making the computation just the same?

Reply:
and/or with the `project_id` field to facilitate the computations. 🤷🏼‍♂️ 

It can be always improved after, for the moment, we just decide to make it simpler possible and during the meeting we got, we choose to use the mixin we used instead of having a analytic distribution. Now, during the implementation, we discover some limitations but it is acceptable but indeed, we think maybe the analytic distribution was maybe the most appropriate but then how we should compute the project profitability?
- Should we take the first analytic account set to project plan? If yes, then how the user could know that? He will more feel this is a issue and report the "bug" but actually, no it is not managed, how can we explain that? 
- Should we take into account all the dimension purposed by the analytic distribution? If we choose that solution then we will have to review completely the project profitability since we could have more than one analytic account for the same analytic plan 
- Should we not take into account the analytic distribution and only use the project field? Then the user cannot use the analytic distribution in SOL and POL to say "hey that SOL, I don't want we take it into account in the project profitability, I would like it will be in another project"

So, let's give a try to that solution and we will see how the users will use that feature, we can always do better after. 🙂 

Your solution could indeed work to have the same behavior than right now, but for me, if we have the analytic distribution field it will not be clear for the user why we don't really take into account the analytic distribution and so, we will probably have many reports because of that. Actually, I am not sure to understand how we could have that constraint to say the main plan (project plan) should be in a line at 100 %. 🧐 

Moreover, we are aware, right now, the user cannot do a complex analytic distribution by default for the documents linked to a project.

Also, that feature has to be done this week (I know it is not an excuse but you can surely understand us that we are a bit in the rush. 😬 ), so if you want to discuss to that, let's create a group on Discord to know if we are in the right direction. The POs in the differents scopes impacted are aware of those changes, unfortunately, they are offs this week and we never mention we should use analytic mixin to have the analytic distribution instead of using the other mixin. 

(it is a bit a bad timing right now for us... 🙁 )

cc @auon-odoo @deneuvillem

Reply:
Me neither. 🙂 

But I did not really see why your solution is simpler.

Reply:
Because it requires about 3000 less lines of diff :smile: